### PR TITLE
Feature PvP Rewards

### LINKED ISSUE #40: Feature PvP Rewards [$50]
**What you can configure?**

- Set amount of myCred points to give
- Set who will get the rewards (Winners or Loosers)
- Set the bracket who will be rewarded (specify one or give to all)
- Month and Year to calculate the total amount.

2 options, a Preview to view a total of 10 values and Send Rewards to give all points.

<bountysource-plugin>

---
There is a **[$50 open bounty](https://www.bountysource.com/issues/98478386-feature-pvp-rewards?utm_campaign=plugin&utm_content=tracker%2F182917716&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F182917716&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by nicoaravena**
New feature Give PvP Rewards (close #40).

**What you can configure?**
- Set amount of myCred points to give
- Set who will get the rewards (Winners or Loosers)
- Set the bracket who will be rewarded (specify one or give to all)
- Month and Year to calculate the total amount.
- Allow to give an extra reward to Top players (5, 10, 15, 20 or 25).

2 options, a Preview to view a total of 10 values and Send Rewards to give all points.

**Limitations:**
- Auth and Character database should be in the same server.

**Impact**
- None.

imagen

### CODE REVIEW

**Comment by nicoaravena** [comment]
@Yehonal just add extra top rewards as talked.

**Comment by Helias** [comment]
I get these errors :S
```
(http://localhost:81/wp-content/plugins/acore-wp-plugin/web/assets/logo.svg): failed to open stream: Cannot assign requested address in /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php on line 75 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-includes/functions.php on line 6362 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-admin/includes/misc.php on line 1310 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-admin/admin-header.php on line 9 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-includes/option.php on line 1051 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-includes/option.php on line 1052
```

**Comment by nicoaravena** [comment]
That's weird, this were added in #32 , did you try giving read permissions to that folder/file?

**Comment by Helias** [comment]
I'll try tomorrow :+1:

File: srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.controller.php [hunk 1]
```diff
@@ -46,21 +46,215 @@ public function loadSettings() {
...
+        $bracket = 0;
+        $bracketAnd = '';
+        $month = 1;
+        $year = 2010;
+        $result = [];
+        $top = 0;
+        $fixedAmount = 0;
+        $stepAmount = 0;
+        // TODO OBTAIN MYCRED FROM DATABASE.
+        $mycredTokenName = "mycred_default";
```
Comment (Helias) [inline srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.controller.php:89]: what about this? this token is exactly `mycred_default`?

Comment (nicoaravena) [reply srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.controller.php]: It was the default mycred value (used in the sample db you sent), I updated it to use the mycred options.

Comment (nicoaravena) [reply srv/wordpress/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/Settings.controller.php]: imagen
This was the default value

**Comment by nicoaravena** [comment]
> 
> 
> I get these errors using this PR :S
> 
> ```
> (http://localhost:81/wp-content/plugins/acore-wp-plugin/web/assets/logo.svg): failed to open stream: Cannot assign requested address in /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php on line 75 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-includes/functions.php on line 6362 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-admin/includes/misc.php on line 1310 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-admin/admin-header.php on line 9 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-includes/option.php on line 1051 Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/wp-content/plugins/acore-wp-plugin/src/components/AdminPanel/AdminPanel.php:75) in /var/www/html/wp-includes/option.php on line 1052
> ```

I fix this in last commit, read svg was using plugin_url instead of the file path.

**Comment by Helias** [comment]
Fixed a little error that I have on `mysql 8.0.25-0ubuntu0.20.04.1`.

**Review by Helias — APPROVED** [review]

**Pull request merged**